### PR TITLE
Remove dependency on custom LLVM

### PR DESCRIPTION
**Pull request by etiotto**
Mark the kernel with the SPIR_KERNEL calling convention and add custom attributes `intel_reqd_sub_group_size` and `max_work_group_size` as passthrough attributes. 

The custom attributes attached to the kernel are post processed after the LLVM IR is generated and transformed into LLVM metadata.

### CODE REVIEW

**Review by pbchekin — APPROVED** [review]

File: scripts/test-triton.sh [hunk 1]
```diff
@@ -60,11 +60,11 @@ export SCRIPTS_DIR=$(dirname "$0")
 python3 -m pip install lit
 python3 -m pip install pytest pytest-xdist pytest-rerunfailures
 
-$SCRIPTS_DIR/compile-pytorch-ipex.sh --pinned $ARGS
-if [ $? -ne 0 ]; then
-  echo "FAILED: return code $?"
-  exit $?
-fi
+#$SCRIPTS_DIR/compile-pytorch-ipex.sh --pinned $ARGS
```
Comment (etiotto) [inline scripts/test-triton.sh:63]: Need to undo this change

File: python/src/llvm.cc [hunk 2]
```diff
@@ -104,10 +104,80 @@ std::string translateLLVMIRToASM(llvm::Module &module,
...
-void findKernels(llvm::Module &M, std::set<llvm::Function *> &functions) {
+static uint32_t findKernels(llvm::Module &M,
+                            std::set<llvm::Function *> &functions) {
+  assert(functions.empty() && "Expecting an empty set");
+  uint32_t numKernels = 0;
   for (llvm::Function &function : M.functions())
-    if (function.getCallingConv() == CallingConv::SPIR_KERNEL)
+    if (function.getCallingConv() == CallingConv::SPIR_KERNEL) {
       functions.insert(&function);
+      numKernels++;
```
Comment (whitneywhtsang) [inline python/src/llvm.cc:114]:
```suggestion
      ++numKernels;
```

**Review by whitneywhtsang — APPROVED** [review]
Instead of building LLVM from source, could try to unset `LLVM_SYSPATH`, which can be done in separate PR.
